tensorflow-utils/src/main/java/org/tensorflow/util/ndarray/impl/dense/DoubleDenseNdArray.java

```diff
@@ -0,0 +1,104 @@
+/*
+ Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ =======================================================================
+ */
+package org.tensorflow.util.ndarray.impl.dense;
+
+import org.tensorflow.util.buffer.DataBuffer;
+import org.tensorflow.util.buffer.DataBuffers;
+import org.tensorflow.util.buffer.DoubleDataBuffer;
+import org.tensorflow.util.ndarray.DoubleNdArray;
+import org.tensorflow.util.ndarray.NdArray;
+import org.tensorflow.util.Shape;
+import org.tensorflow.util.ndarray.impl.dimension.DimensionalSpace;
+
+public class DoubleDenseNdArray extends AbstractDenseNdArray<Double, DoubleNdArray>
+    implements DoubleNdArray {
+
+  public static DoubleNdArray create(DoubleDataBuffer buffer, Shape shape) {
+    Validator.denseShape(buffer, shape);
+    return new DoubleDenseNdArray(buffer, shape);
+  }
+
+  @Override
+  public double getDouble(long... indices) {
+    return buffer().getDouble(positionOf(indices, true));
```

Comment:
Some of these methods go through `buffer()`, some access the `buffer` field directly. It should pick one way, or document that they behave differently (as `DoubleDenseNdArray` isn't final, so could be subclassed).

Reply:
`buffer()` is only there to let the superclass access the `buffer` stored in the subclass. Again, this has changed so many times... I'll sanitize it, thanks!